### PR TITLE
Angular 17

### LINKED ISSUE #7: Error with Angular > 11.0.2
`npm 7` seems to be more strict with dependency resolution and provides the following error when `"@angular/common": "11.2.1"` is installed:

```
npm ERR! Could not resolve dependency:
npm ERR! peer @angular/common@">= 6.0.0 < 11.0.2" from an-qrcode@1.0.7
npm ERR! node_modules/an-qrcode
npm ERR!   an-qrcode@"1.0.7" from the root project
```

A simple solution is probably to bump the https://github.com/naimmalek/an-qrcode/blob/master/package.json#L15 version 🤔

### PR DESCRIPTION
**Pull request by whyboris**
I updated the tests to pass 😎 
Updated all versions of dependencies to ~February 2022~ ~December 2022~ December 2023✅ 

✅  `npm start` works 
✅  `npm test` works 

Happy to make changes if you request 🤝 

Really hoping to close #7 - though unsure if this will be enough 😅 

This PR would also make the PRs #9, #10, #13, #14, #15, #16, #17, #18, #19, #20, #21, #22, & #23 obsolete 😁

### CODE REVIEW

**Comment by whyboris** [comment]
@naimmalek -- please let me know if there's anything you'd like from me for this PR to merge in 🙇

**Comment by whyboris** [comment]
@naimmalek -- could you merge this one in and craft a new release? My app is stuck every time I try to install dependencies 😓  I have to manually delete `"an-qrcode": "1.0.7"` from my `package.json` to run `npm install` without error :trollface:  and then re-add your dependency manually 😅

**Comment by whyboris** [comment]
Hey @naimmalek -- would you be able to merge this in sometime this year? 🙇 

Your repository is really helpful in my project, and it still works 💯 well, but `npm install` keeps erroring out, requiring me to run `npm install --legacy-peer-deps` :trollface:  - a minor inconvenience, but entirely avoidable 😉

**Comment by whyboris** [comment]
I upgraded to `Angular 15` 🥳 
And updated all dependencies to newest versions as of Dec 1st, 2022 😎 
Had to disable 2 lines in [karma.conf.js](https://stackoverflow.com/questions/74546242/karma-jasmine-uncaught-typeerror-webpack-require-context-is-not-a-fu) to get tests to run again 👍 
And had to remove the `relativeLinkResolution` (a deprecated option for routing in Angular) 🤝

**Comment by whyboris** [comment]
@naimmalek -- would you be able to merge this in and create a new release before February 20th? 🙇  - I am hoping to use this repository in my [Video Hub App](https://github.com/whyboris/Video-Hub-App) ❤️ -- I'm aiming to push out a bugfix release on the 5th anniversary of my app 😊

**Comment by whyboris** [comment]
@naimmalek - would you be willing to merge this in? 🤝 
After this I could come back with a cleanup PR for _Angular 17_ 🚀

**Comment by naimmalek** [comment]
@whyboris can you fix conflict to merge this PR

**Comment by whyboris** [comment]
@naimmalek - tried running the code off `master` and I get this error:
```log
- Generating browser application bundles...(node:6440) [DEP0111] DeprecationWarning: Access to process.binding('http_parser') is deprecated.
(Use `node --trace-deprecation ...` to show where the warning was created)
\ Generating browser application bundles (phase: building)...node:internal/crypto/hash:69
  this[kHandle] = new _Hash(algorithm, xofLen);

...

code: 'ERR_OSSL_EVP_UNSUPPORTED'
```
Should `npm start` work off `master` on your repository at this time?

_update_ : meanwhile I updated the code in my PR and `npm start` and `npm test` both work 🥳

File: src/app/app.module.ts [hunk 1]
```diff
@@ -1,6 +1,6 @@
 import { BrowserModule } from '@angular/platform-browser';
 import { NgModule } from '@angular/core';
-import { DeviceDetectorModule } from 'ngx-device-detector';
+// import { DeviceDetectorService } from 'ngx-device-detector';
```
Comment (whyboris) [inline src/app/app.module.ts]: Unclear to me why this is present - it doesn't seem to be used anywhere in the app 🤔 
We can remove it from the `package.json` file I think 👍

**Comment by whyboris** [comment]
- `npm start` works ✅
- `npm build` works ✅ 
- `npm test` works 🤷 but with this error:

```
Error during loading: Uncaught SyntaxError: Unexpected token 'export' in http://localhost:9876/_karma_webpack_/scripts.js line 698
```

Unsure why that's happening, but everything else works with **Angular 17** 🥳

**Comment by whyboris** [comment]
I think it's also worth updating the `tsconfig.json` with:
```
"module": "es2022",
"target": "es2022",
"lib": ["es2022",
```
🤔

**Comment by whyboris** [comment]
@naimmalek 🤝 the PR is ready for merge (please confirm that it works on your end) 👍 

Please let me know if there's anything else I should do 🙇

File: src/app/app.module.ts (hunk 1 above)
Comment (naimmalek) [reply src/app/app.module.ts]: Yes remove from `package.json`  file and remove from all the place (don't comment it).

File: src/test.ts [hunk 2]
```diff
@@ -15,6 +15,6 @@ getTestBed().initTestEnvironment(
   platformBrowserDynamicTesting()
 );
 // Then we find all the tests.
-const context = require.context('./', true, /\.spec\.ts$/);
+// const context = require.context('./', true, /\.spec\.ts$/);
 // And load the modules.
-context.keys().map(context);
+// context.keys().map(context);
```
Comment (naimmalek) [inline src/test.ts:20]: @whyboris why this is commented?

**Review by naimmalek — CHANGES_REQUESTED** [review]

File: src/test.ts (hunk 2 above)
Comment (whyboris) [reply src/test.ts:20]: I believe this code is no longer needed. If I remember right, when you create a new Angular application with the CLI this code is not present 🤔 

Similar code change in a repository that seems to be up to date:
https://github.com/maximegris/angular-electron/commit/680d18df3776617ccaf61bf5f19be5189d938259

File: src/app/app.module.ts (hunk 1 above)
Comment (whyboris) [reply src/app/app.module.ts]: ✅ removed 👍

**Comment by whyboris** [comment]
✅ removed `ngx-device-detector`
✅ removed `postinstall` script as `ngcc` is no longer needed for Angular 🤔 
✅ updated _module_, _target_, and _lib_, in `tsconfig.json` to `es2022` 

The "target" can be lower if you'd like - though I think most people use modern stuff 😅

**Review by naimmalek — APPROVED** [review]

**Pull request merged**